core/src/test/java/org/apache/calcite/util/format/FormatElementEnumTest.java

```diff
@@ -29,8 +29,149 @@
   @Test void testDay() {
+    final String date = "2014-09-30T10:00:00Z";
     StringBuilder ts = new StringBuilder();
-    FormatElementEnum.DAY.format(ts, Date.from(Instant.parse("2014-09-30T10:00:00Z")));
+    FormatElementEnum.DAY.format(ts, Date.from(Instant.parse(date)));
     assertThat(ts, hasToString("Tuesday"));
   }
+
+  @Test void testD() {
+    final String date = "2014-09-30T10:00:00Z";
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.D.format(ts, Date.from(Instant.parse(date)));
+    assertThat(ts, hasToString("3"));
+  }
+
+  @Test void testDD() {
+    final String date = "2014-09-30T10:00:00Z";
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.DD.format(ts, Date.from(Instant.parse(date)));
+    assertThat(ts, hasToString("30"));
+  }
+
+  @Test void testDDD() {
+    final String date = "2014-09-30T10:00:00Z";
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.DDD.format(ts, Date.from(Instant.parse(date)));
+    assertThat(ts, hasToString("273"));
+  }
+
+  @Test void testDY() {
+    final String date = "2014-09-30T10:00:00Z";
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.DY.format(ts, Date.from(Instant.parse(date)));
+    assertThat(ts, hasToString("Tue"));
+  }
+
+  @Test void testFF1() {
+    final String date = "2014-09-30T10:00:00.123456Z";
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.FF1.format(ts, Date.from(Instant.parse(date)));
+    assertThat(ts, hasToString("123"));
+  }
+
+  @Test void testFF2() {
+    final String date = "2014-09-30T10:00:00.123456Z";
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.FF2.format(ts, Date.from(Instant.parse(date)));
+    assertThat(ts, hasToString("123"));
+  }
+
+  @Test void testFF3() {
+    final String date = "2014-09-30T10:00:00.123456Z";
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.FF3.format(ts, Date.from(Instant.parse(date)));
+    assertThat(ts, hasToString("123"));
+  }
+
+  @Test void testFF4() {
+    final String date = "2014-09-30T10:00:00.123456Z";
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.FF4.format(ts, Date.from(Instant.parse(date)));
+    assertThat(ts, hasToString("0123"));
+  }
+
+  @Test void testFF5() {
+    final String date = "2014-09-30T10:00:00.123456Z";
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.FF5.format(ts, Date.from(Instant.parse(date)));
+    assertThat(ts, hasToString("00123"));
+  }
+
+  @Test void testFF6() {
+    final String date = "2014-09-30T10:00:00.123456Z";
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.FF6.format(ts, Date.from(Instant.parse(date)));
+    assertThat(ts, hasToString("000123"));
+  }
+
+  @Test void testIW() {
+    final String date = "2014-09-30T10:00:00Z";
+    StringBuilder ts = new StringBuilder();
+    FormatElementEnum.IW.format(ts, Date.from(Instant.parse(date)));
+    assertThat(ts, hasToString("40"));
+  }
+
```

Comment:
There are too many duplicate codes in the test, which can be optimized and merged.